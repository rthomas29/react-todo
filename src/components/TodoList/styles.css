.list-container {
  display: flex;
  flex-flow: column wrap;
  margin-top: 5px;
}

.input {
  height: 25px;
}

input::placeholder {
  font-size: 1.2em;
}

.list-item {
  display: flex;
  flex-flow: row nowrap;
}

.complete {
  text-decoration: line-through;
}

.clear-btn {
  height: 31px;
  border-radius: 5px;
  background-color: #26a65b;
  color: white;
  font-size: 1em;
}

.delete-btn {
  height: 31px;
  border-radius: 5px;
  background-color: #f22613;
  color: white;
  font-size: 1em;
}

.btn-group {
  display: flex;
  flex-flow: row nowrap;
  max-width: 300;
}