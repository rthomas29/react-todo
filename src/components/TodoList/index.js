import React from 'react'
import './styles.css'

const TodoList = ({ todos, handleTaskDelete, toggleTaskCompletion }) => (
  <main className='list-container'>
    {todos.map((todo, i) => {
      const listNumber = i + 1
      const key = `${i}:${todo.task}`
      const style = todo.isComplete ? { textDecoration: 'line-through' } : {}
      return (
        <div
          key={key}
          className='list-container'
        >
          <div className='list-item'>
            <input type='checkbox' className='form-check-input' checked={todo.isComplete} onClick={() => toggleTaskCompletion(i)} />
            <p style={style}>
              {`${listNumber}. ${todo.task}`}
            </p>
          </div>
        </div>
      )
    })}
  </main>
)

export default TodoList