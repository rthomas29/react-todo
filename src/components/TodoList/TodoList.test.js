import React from 'react'
import { render } from 'react-testing-library'
import TodoList from './index'

describe('<TodoList />', () => {
  it('should match snapshot', () => {
    const todos = [{ task: 'task 1', isComplete: false }, { task: 'task 2', isComplete: false }]
    const { container } = render(<TodoList todos={todos} />)
    expect(container.firstChild).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })
})