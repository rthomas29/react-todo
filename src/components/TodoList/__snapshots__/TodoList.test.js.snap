// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoList /> should match snapshot 1`] = `
<main
  class="list-container"
>
  <div
    class="list-container"
  >
    <div
      class="list-item"
    >
      <input
        class="form-check-input"
        type="checkbox"
      />
      <p>
        1. task 1
      </p>
    </div>
  </div>
  <div
    class="list-container"
  >
    <div
      class="list-item"
    >
      <input
        class="form-check-input"
        type="checkbox"
      />
      <p>
        2. task 2
      </p>
    </div>
  </div>
</main>
`;
